src/main/kotlin/com/coder/toolbox/CoderToolboxContext.kt

```diff
@@ -36,14 +36,15 @@ data class CoderToolboxContext(
      *
      * In order of preference:
      *
-     * 1. Last used URL.
-     * 2. URL in settings.
-     * 3. CODER_URL.
-     * 4. URL in global cli config.
+     * 1. Last used URL from the settings.
+     * 2. Last used URL from the secrets store.
+     * 3. Default URL
      */
     val deploymentUrl: URL
         get() {
-            if (this.secrets.lastDeploymentURL.isNotBlank()) {
+            if (!this.settingsStore.lastDeploymentURL.isNullOrBlank()) {
+                return this.settingsStore.lastDeploymentURL!!.toURL()
```

Comment:
Is this technically unsafe (the timing would have to be insane though)?  Or is it actually impossible for this value to change between these two lines?  I just get nervous when I see `!!` lol.

Reply:
Theoretically you can have two threads racing - one changing the value of lastDeploymentURL and another one reading. In our code that should be pretty much impossible. But the compiler can be sure of that so it throws and error. One common approach is to just hold the result in a local variable. Once we checked for null safety then !! is no longer needed.